src/Test.UnitTests.Sarif/Core/SarifLogTests.cs

```diff
@@ -200,6 +204,152 @@
             newSarifLog.Runs[0].Results.Count.Should().Be(sarifLog.Runs[0].Results.Count);
         }
 
+        [Fact]
+        public async Task SarifLog_PostStream_WithInvalidParameters_ShouldThrowArgumentNullException()
```

Comment:
>SarifLog_PostStream_WithInvalidParameters_ShouldThrowArgumentNullException [](http://example.com/codeflow?start=26&length=74)

I don't understand this test pattern. 

Why aren't we simply using the helpers that verify an assertion is thrown?

All this test goo, creating titles, etc, isn't helpful for this simple scenario, it's a lot of complexity for not a lot of value.

Basically, all these tests could be authored by updating the code below to simply to pass parameters.

                Exception exception = await Record.ExceptionAsync(async () =>
                {
                    await SarifLog.Post(testCase.PostUri,
                                        testCase.Stream,
                                        testCase.HttpClient);
                });

Have you taken a look at the actual test output produced by this pattern? It's not great. :) That's another reason why we prefer not to use it unless necessary.

Reply:
replaced to Assert.Throw style.